tests/v2/testlib/wakunode.nim

```diff
@@ -0,0 +1,73 @@
+import
+  std/options,
+  stew/results,
+  stew/shims/net,
+  chronos,
+  libp2p/switch,
+  libp2p/builders,
+  libp2p/nameresolving/nameresolver,
+  libp2p/crypto/crypto as libp2p_keys,
+  eth/keys as eth_keys
+import
+  ../../../waku/v2/waku_node,
+  ../../../waku/v2/node/peer_manager,
+  ../../../waku/v2/protocol/waku_enr,
+  ../../../waku/v2/protocol/waku_discv5,
+  ./common
+
+
+# Waku node
+
+proc newTestWakuNode*(nodeKey: crypto.PrivateKey,
+                      bindIp: ValidIpAddress,
+                      bindPort: Port,
+                      extIp = none(ValidIpAddress),
+                      extPort = none(Port),
+                      extMultiAddrs = newSeq[MultiAddress](),
+                      peerStorage: PeerStorage = nil,
+                      maxConnections = builders.MaxConnections,
+                      wsBindPort: Port = (Port)8000,
+                      wsEnabled: bool = false,
+                      wssEnabled: bool = false,
+                      secureKey: string = "",
+                      secureCert: string = "",
+                      wakuFlags = none(CapabilitiesBitfield),
+                      nameResolver: NameResolver = nil,
+                      sendSignedPeerRecord = false,
+                      dns4DomainName = none(string),
+                      discv5UdpPort = none(Port),
+                      wakuDiscv5 = none(WakuDiscoveryV5),
+                      agentString = none(string),
+                      peerStoreCapacity = none(int)): WakuNode =
+  let netConfigRes = NetConfig.init(
+    bindIp = bindIp,
+    bindPort = bindPort,
+    extIp = extIp,
+    extPort = extPort,
+    extMultiAddrs = extMultiAddrs,
+    wsBindPort = wsBindPort,
+    wsEnabled = wsEnabled,
+    wssEnabled = wssEnabled,
+    wakuFlags = wakuFlags,
+    dns4DomainName = dns4DomainName,
+    discv5UdpPort = discv5UdpPort,
+  )
+  if netConfigRes.isErr:
+    raise newException(Defect, "Invalid network configuration")
```

Comment:
Shall we have $netConfigRes.error appended in this point?

Also, I wonder if we could have a future PR where we convert all the "raise exception" approaches into Result[...] ones.

Reply:
> Also, I wonder if we could have a future PR where we convert all the "raise exception" approaches into Result[...] ones.

The problem is how the node instance creation, the `newTestWakuNode()`, is used around the test suites. The test cases expect either the `WakuNode.new()` method to succeed and return the instance or raise an exception. Changing the API for a result would mean many more complex changes in the test suites' codebase.

I defend that, in general, `Result`s should prevail over exceptions. This is true for "production code". 

In this case, these changes are on the test suites codebase, so it is ok to raise a `Defect` (the equivalent of a `panic` in Rust/Go) or an exception if there is any issue with the instantiation of the node as it will halt the test case and it will mark it as failed.